javatests/arcs/core/host/ArcHostManagerTest.kt

```diff
@@ -16,30 +12,29 @@ import org.junit.runners.JUnit4
 @OptIn(ExperimentalCoroutinesApi::class)
 open class ArcHostManagerTest {
 
-  @Before
-  fun setUp() = runBlocking<Unit> {
-    RamDisk.clear()
-    DriverAndKeyConfigurator.configure(null)
-  }
-
   @Test
   fun pauseAllHostsFor() = runBlocking {
     val schedulerProvider = SimpleSchedulerProvider(coroutineContext)
-    val host = TestHost(schedulerProvider)
+    val hosts = listOf(
+      PausableTestingHost(schedulerProvider),
+      PausableTestingHost(schedulerProvider),
+      PausableTestingHost(schedulerProvider)
+    )
     val hostRegistry = ExplicitHostRegistry()
-    hostRegistry.registerHost(host)
+    hosts.forEach { hostRegistry.registerHost(it) }
 
-    val partition = Plan.Partition("arcId", "arcHost", listOf())
-    host.startArc(partition)
-    assertThat(host.lookupArcHostStatus(partition)).isEqualTo(ArcState.Running)
+    assertThat(PausableTestingHost.numPauses).isEqualTo(0)
+    assertThat(PausableTestingHost.numUnpauses).isEqualTo(0)
 
-    var stateDuringPause: ArcState? = null
     ArcHostManager.pauseAllHostsFor {
-      stateDuringPause = host.lookupArcHostStatus(partition)
+      // All hosts have been paused
+      assertThat(PausableTestingHost.numPauses).isEqualTo(3)
+      assertThat(PausableTestingHost.numUnpauses).isEqualTo(0)
```

Comment:
Using the `isPaused` member variable idea, this would become:

```
hosts.forEach {
  assertThat(it.isPaused).isTrue()
}
```

You would also want to add something here that will let you assert that the `pauseAllHostsFor` callback was actually run, e.g.

```
var callbackWasRun = false
ArcHostManager.pauseAllHostsFor {
  callbackWasRun = true
  hosts.forEach {
    assertThat(it.isPaused).isTrue()
  }
}

assertThat(callbackWasRun).isTrue()
hosts.forEach {
  assertThat(it.isPaused).isFalse()
}
```